hooks/boot/06-postgresql-upgrade-extensions.rb

```diff
@@ -41,6 +33,20 @@ def postgresql_upgrade(new_version)
 
     logger.notice("Migrating PostgreSQL data")
 
+    # We need to know the collation and ctype of the DB as postgresql-setup --upgrade
+    # doesn't detect those on its own: https://issues.redhat.com/browse/RHEL-58410
+    # dnf switch-to has already replaced the packages and the cluster is down
+    # We have to start the cluster in single-user mode with the old postgres binary
+    postgres_result = `echo "select datcollate,datctype from pg_database where datname='postgres';" | runuser -l postgres -c '/usr/lib64/pgsql/postgresql-#{current_version}/bin/postgres --single -D /var/lib/pgsql/data postgres'`
+    data = {}
+    postgres_result.each_line do |line|
+      line.match(/dat(collate|ctype)\s+=\s+(\S+)/) do |m|
+        data[m[1]] = m[2].delete('"')
+      end
+    end
```

Comment:
What an ewwww block.

Reply:
Using `/dat(collate|ctype)\s+=\s+"(\S+)"/` as the regex removes the need for `.delete('"')`, making it more ew than ewwww.

I do wonder about error handling: what if `data` doesn't contain any values because the process failed to start up?

Reply:
Then the upgrade will fail as we pass invalid data in `PGSETUP_INITDB_OPTIONS` -- nothing new compared to how it was before this whole hack :see_no_evil: